Small Spec Improvements (#3084)
modelcontextprotocol/modelcontextprotocol: docs/specification/draft/basic/index.mdx

- Error responses **MUST** include the same ID as the request they correspond to (except in error cases where the ID could not be read due a malformed request).
- Error responses **MUST** include an `error` field with a `code` and `message`.
- Error codes **MUST** be integers.

> - Error responses **MAY** include additional information of any type, for example detailed error informationo or nested errors.

Reviewer: ```suggestion
- Error responses **MAY** include a `data` member with additional information of any type, such
  as nested errors.
```

### `_meta`

The `_meta` property/parameter is reserved by MCP to allow clients and servers
The `_meta` property/parameter is used by MCP to allow clients and servers
to attach additional metadata to their interactions.

Certain key names are reserved by MCP for protocol-level metadata, as specified below;
implementations MUST NOT make assumptions about values at these keys.

> Additionally, definitions in the [schema](https://github.com/modelcontextprotocol/specification/blob/main/schema/draft/schema.ts)

Reviewer: In the schema, we [still have](https://github.com/modelcontextprotocol/modelcontextprotocol/blob/main/schema/draft/schema.ts#L67C3-L67C16) `progressToken`, so we have a custom reserved name. I think it's fine, since we say "certain key names are reserved" but I think this part was likely referring to fields like that.
